pkg/addonmanager/controllers/cmamanagedby/controller.go

```diff
@@ -92,3 +100,42 @@ func (c *cmaManagedByController) sync(ctx context.Context, syncCtx factory.SyncC
 	_, err = c.addonPatcher.PatchLabelAnnotations(ctx, cmaCopy, cmaCopy.ObjectMeta, cma.ObjectMeta)
 	return err
 }
+
+func (c *cmaManagedByController) handleFinalizer(ctx context.Context, cma *addonapiv1alpha1.ClusterManagementAddOn, addonName string) error {
```

Comment:
I think this method can be simplified to if there is one or more ManagedClusterAddOn objects, ensure the finalizer on the CMA, otherwise, remove the finalizer.